src/httpclient.c

```diff
@@ -114,7 +118,26 @@ void httpclient_timeout_config(
     if (curl_easy_setopt(curl, CURLOPT_CONNECTTIMEOUT_MS, cto) != CURLE_OK) {
         corto_error("Failed to set CURLOPT_CONNECTTIMEOUT_MS.");
     }
+}
+
+void httpclient_auth_config(
+    CURL *curl)
+{
+    corto_string user = httpclient_get_user();
```

Comment:
Suggest to make a function that returns the `httpclient_Config` object instead of the individual members. That saves you from having to do multiple TLS lookups, which while they are fast, aren't free.